### ISSUE TITLE
"You have not included any fields in your question text" ERROR

### ISSUE THREAD

**Issue by lucasyanetsko** [marcusgreen/moodle-qtype_gapfill#94]
![Screen Shot 2022-09-21 at 11 24 11 AM](https://user-images.githubusercontent.com/89038943/191545279-bf518d8d-eb12-47c1-8ac0-99a468be5837.png)

When I hit the gray "Edit Question Text" button it's not recognizing the brackets with the containing text.  It's also displaying a host of errors at the top of the screen.

![Screen Shot 2022-09-21 at 11 26 27 AM](https://user-images.githubusercontent.com/89038943/191545876-6f1a4924-608c-4a62-bf7d-ef333b6ea42b.png)

**Comment by marcusgreen** [comment]
Could you export the question to xml and attach it to a post here

**Comment by lucasyanetsko** [comment]
Hi Marcus,

Really appreciate the prompt response.  Below is a the xml from the question.

<p dir="ltr" style="text-align: left;">Charlotte is in [North] [Carolina]</p>

Thank you

**Comment by marcusgreen** [comment]
Sorry I should have been more explicit, I was expecting the actual xml exported which will contain additional information such as the moodle version and the question type version (though you might just be able to give me that).
See attached file and the example from it is this 
    
Though it looks to me like there is something missing from the data in your question. You could try importing the xml attached and see what happens.

<?xml version="1.0" encoding="UTF-8"?>
<quiz>

  <question type="category">
    <category>
      <text>$course$/top/Default for GQT/Exports</text>
    </category>
    <info format="html">
      <text></text>
    </info>
    <idnumber></idnumber>
  </question>


  <question type="gapfill">
    <name>
      <text>Where is Charlotte</text>
    </name>
    <questiontext format="html">
      <text><![CDATA[<p dir="ltr" style="text-align: left;">Charlotte is in [North] [Carolina]</p>]]></text>
    </questiontext>
    <generalfeedback format="html">
      <text></text>
    </generalfeedback>
    <defaultgrade>2.0000000</defaultgrade>
    <penalty>0.3333333</penalty>
    <hidden>0</hidden>
    <idnumber></idnumber>
    <answerdisplay>dragdrop</answerdisplay>
    <delimitchars>[]</delimitchars>
    <casesensitive>0</casesensitive>
    <noduplicates>0</noduplicates>
    <disableregex>0</disableregex>
    <fixedgapsize>1</fixedgapsize>
    <optionsaftertext>0</optionsaftertext>
    <letterhints>1</letterhints>
    <singleuse>1</singleuse>
    <answer fraction="100" format="moodle_auto_format">
      <text>North</text>
      <feedback format="moodle_auto_format">
        <text></text>
      </feedback>
    </answer>
    <answer fraction="100" format="moodle_auto_format">
      <text>Carolina</text>
      <feedback format="moodle_auto_format">
        <text></text>
      </feedback>
    </answer>
    <delimitchars>[]</delimitchars>
    <answerdisplay>dragdrop</answerdisplay>
    <casesensitive>0</casesensitive>
    <noduplicates>0</noduplicates>
    <disableregex>0</disableregex>
    <fixedgapsize>1</fixedgapsize>
    <optionsaftertext>0</optionsaftertext>
    <letterhints>1</letterhints>
      <gapsetting>
        <question>4697</question>
        <gaptext>North</gaptext>
        <itemid>id1_0</itemid>
        <correctfeedback><![CDATA[Yes north is correct]]></correctfeedback>
        <incorrectfeedback><![CDATA[No that is not correct]]></incorrectfeedback>
     </gapsetting>
      <gapsetting>
        <question>4697</question>
        <gaptext>Carolina</gaptext>
        <itemid>id2_0</itemid>
        <correctfeedback><![CDATA[Yes Carolina is correct]]></correctfeedback>
        <incorrectfeedback><![CDATA[That is not correct]]></incorrectfeedback>
     </gapsetting>
    
    <correctfeedback format="html">
      <text>Your answer is correct.</text>
    </correctfeedback>
    <partiallycorrectfeedback format="html">
      <text>Your answer is partially correct.</text>
    </partiallycorrectfeedback>
    <incorrectfeedback format="html">
      <text>Your answer is incorrect.</text>
    </incorrectfeedback>
  </question>

</quiz>

**Comment by marcusgreen** [comment]
My apologies, I should have been more explicit. I meant for you to export the actual xml which will contain additional information such as the following.

``<!-- Gapfill release:2.134 version:2022061900 Moodle version:2021051704.11 release:3.11.4+ (Build: 20211217) -->``

Though it looks like your question is missing some data.
Here is the full xml

[carolina.zip](https://github.com/marcusgreen/moodle-qtype_gapfill/files/9619566/carolina.zip)

**Comment by marcusgreen** [comment]
I spent a summer in North Carolina once

**Comment by lucasyanetsko** [comment]
Thanks Marcus!  Awesome, yeah the summers here can get pretty warm haha where are you located now?

I can't actually export it as the errors are causing me to be unable to save it.

In one of our dev environments version 2.133 is working correctly but 2.134 seems to be spitting out the errors.  Would this potentially have something to do with it?

**Comment by marcusgreen** [comment]
Per Gap feedback was introduced a while ago...
Version 1.96 of the Moodle Gapfill question type Oct 2017
(From https://github.com/marcusgreen/moodle-qtype_gapfill/blob/master/changelog.md)

The errors you are seeing such as undefined property $stdClass::delimitchars  are unrelated to the per gapfeedback. Are you able to create new simple questions on that site or is it an only an issue with existing questions?

The summer I spent was 1983, Am now in (Old) York

**Comment by marcusgreen** [comment]
Can you paste in the error message (currently I can see an image)

**Comment by lucasyanetsko** [comment]
Yes I cannot create a simple new question and am getting an error below the Question text box that reads:

You have not included any fields in your question text

And this is the image of it 
![Screen Shot 2022-09-21 at 4 12 05 PM](https://user-images.githubusercontent.com/89038943/191601138-a0965cf1-140d-41fb-a4ce-8d48c7820fad.png)

**Comment by marcusgreen** [comment]
I am not sure what is happening. The message that says "You have not included any fields in your question text"  suggest it is not seeing any delimiting characters which default to an opening [ and a closing ]. Can you experiment with changing them as a test, e.g. { }  and adjusting the text to match. That doesn't make any sense but I cannot think what the issue might be.
